mmdet/models/anchor_heads/yolact_head.py

```diff
@@ -0,0 +1,601 @@
+            cls_feat = cls_conv(cls_feat)
+        for box_conv in self.box_convs:
+            box_feat = box_conv(box_feat)
+        cls_score = self.yolact_cls(cls_feat)
+        bbox_pred = self.yolact_reg(box_feat)
+        for coeff_conv in self.coeff_convs:
+            coeff_feat = coeff_conv(coeff_feat)
+        coeff_pred = self.yolact_coeff(coeff_feat)
+        coeff_pred = torch.tanh(coeff_pred)
+        return cls_score, bbox_pred, coeff_pred
+
+    @force_fp32(apply_to=('cls_scores', 'bbox_preds'))
+    def loss(self,
+             cls_scores,
+             bbox_preds,
+             gt_bboxes,
+             gt_labels,
+             img_metas,
+             cfg,
+             gt_bboxes_ignore=None,
+             return_sampling_results=False):
+        featmap_sizes = [featmap.size()[-2:] for featmap in cls_scores]
+
+        device = cls_scores[0].device
+
+        anchor_list, valid_flag_list = self.get_anchors(
+            featmap_sizes, img_metas, device=device)
+
+        label_channels = self.cls_out_channels
+
+        cls_reg_targets = self.yolact_target(
+            anchor_list,
+            valid_flag_list,
+            gt_bboxes,
+            img_metas,
+            self.target_means,
+            self.target_stds,
+            cfg,
+            gt_bboxes_ignore_list=gt_bboxes_ignore,
+            gt_labels_list=gt_labels,
+            label_channels=label_channels,
+            sampling=False,
+            unmap_outputs=False,
+            return_sampling_results=return_sampling_results)
+        if return_sampling_results:
+            (labels_list, label_weights_list, bbox_targets_list,
+             bbox_weights_list, num_total_pos, num_total_neg,
+             sampling_results) = cls_reg_targets
+        else:
+            (labels_list, label_weights_list, bbox_targets_list,
+             bbox_weights_list, num_total_pos, num_total_neg) = cls_reg_targets
+        num_total_samples = num_total_pos
+        losses_cls, losses_bbox = multi_apply(
+            self.loss_single,
+            cls_scores,
+            bbox_preds,
+            labels_list,
+            label_weights_list,
+            bbox_targets_list,
+            bbox_weights_list,
+            num_total_samples=num_total_samples,
+            cfg=cfg)
+        if return_sampling_results:
+            return dict(
+                loss_cls=losses_cls, loss_bbox=losses_bbox), sampling_results
+        return dict(loss_cls=losses_cls, loss_bbox=losses_bbox)
+
+    def loss_single(self, cls_score, bbox_pred, labels, label_weights,
+                    bbox_targets, bbox_weights, num_total_samples, cfg):
+        labels = labels.reshape(-1)
+        label_weights = label_weights.reshape(-1)
+        cls_score = cls_score.permute(0, 2, 3,
+                                      1).reshape(-1, self.cls_out_channels)
+        loss_cls_all = F.cross_entropy(
+            cls_score, labels, reduction='none') * label_weights
```

Comment:
It may be better to make the loss configurable like [here](https://github.com/open-mmlab/mmdetection/blob/7198451afced96091a817a31487ee5038a303aa2/mmdet/models/anchor_heads/anchor_head.py#L43).

Reply:
Because we use OHEM, I followed the [code](https://github.com/open-mmlab/mmdetection/blob/7198451afced96091a817a31487ee5038a303aa2/mmdet/models/anchor_heads/ssd_head.py#L113) in SSD. I'll leave it unchanged for now.